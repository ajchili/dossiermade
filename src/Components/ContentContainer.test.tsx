import * as React from "react";
import ContentContainer from "./ContentContainer";
import { shallow } from "enzyme";

describe("ContentContainer", () => {
  it("sets the parent div id to the specified id", () => {
    const tree = shallow(<ContentContainer id="123" cardContent={<div />} />);
    expect(tree).toMatchSnapshot();
  });

  it("renders provided cardContent", () => {
    const tree = shallow(
      <ContentContainer id="" cardContent={<h3>Content</h3>} />
    );
    expect(tree).toMatchSnapshot();
  });

  it("renders a light card by default", () => {
    const tree = shallow(<ContentContainer id="" cardContent={<div />} />);
    expect(tree).toMatchSnapshot();
  });

  it("renders a light card when specified", () => {
    const tree = shallow(
      <ContentContainer id="" cardContent={<div />} backgroundColor="light" />
    );
    expect(tree).toMatchSnapshot();
  });

  it("renders a dark card when specified", () => {
    const tree = shallow(
      <ContentContainer id="" cardContent={<div />} backgroundColor="dark" />
    );
    expect(tree).toMatchSnapshot();
  });
});
