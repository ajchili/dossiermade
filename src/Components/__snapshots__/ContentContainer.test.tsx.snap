// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentContainer renders a dark card when specified 1`] = `
<div
  className="uk-background-secondary"
  id=""
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "marginLeft": "15%",
        "marginRight": "15%",
        "paddingBottom": "7.5vh",
        "paddingTop": "7.5vh",
      }
    }
  >
    <div />
  </div>
</div>
`;

exports[`ContentContainer renders a light card by default 1`] = `
<div
  className="uk-background-muted"
  id=""
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "marginLeft": "15%",
        "marginRight": "15%",
        "paddingBottom": "7.5vh",
        "paddingTop": "7.5vh",
      }
    }
  >
    <div />
  </div>
</div>
`;

exports[`ContentContainer renders a light card when specified 1`] = `
<div
  className="uk-background-muted"
  id=""
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "marginLeft": "15%",
        "marginRight": "15%",
        "paddingBottom": "7.5vh",
        "paddingTop": "7.5vh",
      }
    }
  >
    <div />
  </div>
</div>
`;

exports[`ContentContainer renders provided cardContent 1`] = `
<div
  className="uk-background-muted"
  id=""
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "marginLeft": "15%",
        "marginRight": "15%",
        "paddingBottom": "7.5vh",
        "paddingTop": "7.5vh",
      }
    }
  >
    <h3>
      Content
    </h3>
  </div>
</div>
`;

exports[`ContentContainer sets the parent div id to the specified id 1`] = `
<div
  className="uk-background-muted"
  id="123"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "marginLeft": "15%",
        "marginRight": "15%",
        "paddingBottom": "7.5vh",
        "paddingTop": "7.5vh",
      }
    }
  >
    <div />
  </div>
</div>
`;
